### PR TITLE
Add django-template-partials example to Tips

### PR DESCRIPTION
**Pull request by carltongibson**
Hey @adamchainz 👋 

I saw on _the Fedi_ that you might accept an example using django-template-partials to the tips section, so I drafted one.

### CODE REVIEW

**Comment by adamchainz** [comment]
This project uses present tense PR titles/commit messages without a full-stop at the end, because that makes grammatical sense for a headline 😉

**Comment by adamchainz** [comment]
I like the start here, but I think we should have two sections. One to cover the “render a small portion of the page” style that django-template-partials is perfect for, and another for the “render the only main section” style that the current tip covers. I wrote the current tip after several people asked for it, but I think the django-template-partials style is more in-keeping with the htmx philosophy.

WDYT?

**Comment by carltongibson** [comment]
Hey @adamchainz — It's your package, so happy to follow your lead. 😀

From what you wrote I can't quite work out if you mean: 

1. Have two examples using django-template-partials for the _small snippet_ and _just the main content_ (so add a second example to this PR) or... 
2. Replace the example I added here with just a _small snippet_ case? 

FWIW I'm using _dtp_ for both IRL, but I'd rather add the example(s) you'd be happiest with here. 

LMK and I'll update. 🎁

**Comment by adamchainz** [comment]
Coming back to django-htmx after a bit of a break. I think perhaps let’s change to showcase only django-template-partials, especially since you’ll now be getting something similar into Django itself. Let’s make these changes:

1. Update the “partial rendering” demo in the example app to use django-template-partials, probably using a second view to render the table partial.
2. Split the tips into two: “Partial rendering of a section” and ”Partial rendering of main content”.
3. In “Partial rendering of a section”, cover the demo from the example project.
4. In “Partial rendering of main content”, put the tip as you’ve contributed here. Put the intro paragraph from the old tip: “For requests made with htmx...”.
5. Preserve the existing content, with a warning about it being more involved and a kind-of legacy method, under “Partial rendering of main content with template inheritance”.

What do you think?

**Comment by carltongibson** [comment]
Hey @adamchainz. Makes sense. Give me a little cycle and I'll get that done. 👍

File: docs/tips.rst [hunk 1]
```diff
@@ -104,3 +104,50 @@ Here, ``_base.html`` would be the main site base:
...
+
+There's a third-party package called `django-template-partials`_ that allows defining reusable named inline partials for the Django Template Language.
+
+Once you've installed ``django-template-partials`` you can add the ``{% partialdef %}`` tag to your template to mark the reusable section:
+
+.. code-block:: django
+
+   {% extends _base.html %}
+   {% block main %}
+      {% partialdef main inline %}
```
[the remark this answers is not in the document]
Comment (carltongibson) [reply docs/tips.rst:120]: https://github.com/adamchainz/django-htmx/pull/413/commits/64d58cab18bae23b5011be29ee88878dfed798c6

**Comment by carltongibson** [comment]
@adamchainz I've made the changes in the example project. I updated the existing view. Do you think it's worth duplicating what's there to have both methods runnable? 

Will move to docs next.

**Comment by carltongibson** [comment]
Hey @adamchainz. Summer 🏖️

I've made what I hope is an OK pass at the suggested changes. If you could glance at them when you can, I can make any adjustments you want, and then squash down etc. 

Thanks.

**Comment by adamchainz** [comment]
Okay, finally, there we go! Thank you 💚

**Comment by carltongibson** [comment]
Slowly slowly catchy monkey 🐵 

Thanks Adam.

**Pull request merged**